internal/cli/api/api.go

```diff
@@ -447,8 +463,42 @@ func ensureVersionIsSupported(apiCommand shared_api.Command, version *string) {
 		return
 	}
 
-	fmt.Fprintf(os.Stderr, "warning: version '%s' is not supported for this endpoint, using default API version '%s'; consider pinning a version to ensure consisentcy when updating the CLI\n", *version, defaultVersion)
-	*version = defaultVersion
+	fmt.Fprintf(os.Stderr, "warning: version '%s' is not supported for this endpoint, using default API version '%s'; consider pinning a version to ensure consisentcy when updating the CLI\n", *versionString, defaultVersion)
+	*versionString = defaultVersion
+}
+
+func printPreviewWarning(apiCommand shared_api.Command, versionString *string) {
+	version, err := shared_api.ParseVersion(*versionString)
+
+	// If the version is invalid return, this should never happen
+	if err != nil {
+		return
+	}
```

Comment:
Similar to the above comment, we should at least log the error to stderror in case the impossible happen and we need to debug the issue. If you don't agree, you could then remove this if and have `version, _ := shared_api.ParseVersion(*versionString)`

Reply:
Thanks for the suggestion, I've updated these branches with an error message!